appgate/resource_appgate_appliance_controller.go

```diff
@@ -0,0 +1,253 @@
+package appgate
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/appgate/sdp-api-client-go/api/v16/openapi"
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAppgateApplianceControllerActivation() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAppgateApplianceControllerActivationCreate,
+		ReadContext:   resourceAppgateApplianceControllerActivationRead,
+		UpdateContext: resourceAppgateApplianceControllerActivationUpdate,
+		DeleteContext: resourceAppgateApplianceControllerActivationDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(15 * time.Minute),
+			Update: schema.DefaultTimeout(15 * time.Minute),
+			Delete: schema.DefaultTimeout(15 * time.Minute),
+		},
```

Comment:
Assuming these timeouts mean the new controller is fully joined (the database was fully synced) and operational. Enabling the controller role can take longer than this depending on the database size and latency. E.g. if you have controllers on the opposite side of the world with a fairly big database. For most cases it is more than enough though.

These are just defaults right?

Reply:
yes, this is default values can take be overwritten in the resource.
https://www.terraform.io/docs/language/resources/syntax.html#operation-timeouts
for example
```hcl
resource "appgatesdp_appliance_controller_activation" "activate_second_controller" {
  timeouts {
    create = "60m"
    delete = "2h"
  }
  appliance_id = "69723c1d-0b5f-44f8-9f29-3eed57618440"
  controller {
    enabled = true
  }
  admin_interface {
    hostname   = "envy-10-97-170-3.devops"
    https_port = 8443
    https_ciphers = [
      "ECDHE-RSA-AES256-GCM-SHA384",
      "ECDHE-RSA-AES128-GCM-SHA256"
    ]
  }
}

```